nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/flow/synchronization/StandardVersionedComponentSynchronizer.java

```diff
@@ -1433,6 +1434,14 @@ private Map<String, String> populatePropertiesMap(final ComponentNode componentN
                         final String serviceVersionedComponentId = proposedProperties.get(propertyName);
                         String instanceId = getServiceInstanceId(serviceVersionedComponentId, group);
                         value = (instanceId == null) ? serviceVersionedComponentId : instanceId;
+
+                        // Find the same property descriptor in the component's CreatedExtension and replace it with the
+                        // instance ID of the service
+                        createdExtensions.stream().filter(ce -> ce.extension.equals(componentNode)).forEach(createdExtension -> {
+                            LOG.debug("Replacing CreatedExtension property {} old value {} with new value {}",
+                                    propertyName, createdExtension.propertyValues.get(propertyName) , value);
```

Comment:
This debug log should be removed because property values may contain sensitive information that should not be logged.
```suggestion
```

Reply:
Good point @exceptionfactory, I have removed the debug log statement.